guides/keras_cv/classification_with_keras_cv.py

```diff
@@ -0,0 +1,701 @@
+For many tasks, no relevant pretrained model
+will be available (e.g., categorizing images specific to your application).
+
+First, let's get started by loading some data:
+"""
+
+BATCH_SIZE = 32
+IMAGE_SIZE = (224, 224)
+AUTOTUNE = tf.data.AUTOTUNE
+tfds.disable_progress_bar()
+
+data, dataset_info = tfds.load("cats_vs_dogs", with_info=True, as_supervised=True)
+train_steps_per_epoch = dataset_info.splits["train"].num_examples // BATCH_SIZE
+train_dataset = data["train"]
+
+num_classes = dataset_info.features["label"].num_classes
+
+resizing = keras_cv.layers.Resizing(
+    IMAGE_SIZE[0], IMAGE_SIZE[1], crop_to_aspect_ratio=True
+)
+
+
+def preprocess_inputs(image, label):
+    image = tf.cast(image, tf.float32)
+    return resizing(image), tf.one_hot(label, num_classes)
+
+
+train_dataset = train_dataset.shuffle(10 * BATCH_SIZE).map(
+    preprocess_inputs, num_parallel_calls=AUTOTUNE
+)
+train_dataset = train_dataset.batch(BATCH_SIZE)
+
+images = next(iter(train_dataset.take(1)))[0]
+keras_cv.visualization.plot_image_gallery(images, value_range=(0, 255))
+
+"""
+Meow!
+
+Next let's construct our model:
+"""
+
+backbone = keras_cv.models.EfficientNetV2Backbone.from_preset(
+    "efficientnetv2_b0_imagenet",
+)
+"""
+The use of imagenet in the preset name indicates that the backbone was
+pretrained on the ImageNet dataset.
+Pretrained backbones extract more information from our labeled examples by
+leveraging patterns extracted from potentially much larger datasets.
+
+Next lets put together our classifier:
+"""
+
+model = keras.Sequential(
+    [
+        backbone,
+        keras.layers.GlobalMaxPooling2D(),
+        keras.layers.Dropout(rate=0.5),
+        keras.layers.Dense(2, activation="softmax"),
+    ]
+)
+model.compile(
+    loss="categorical_crossentropy",
+    optimizer=tf.optimizers.SGD(learning_rate=0.01),
+    metrics=["accuracy"],
+)
+
+"""
+Here our classifier is just a simple `keras.Sequential`.
+All that is left to do is call `model.fit()`:
+"""
+
+model.fit(train_dataset)
+
+
+"""
+Let's look at how our model performs after the fine tuning:
+"""
+
+predictions = model.predict(np.expand_dims(image, axis=0))
+
+classes = {0: "cat", 1: "dog"}
+print("Top class is:", classes[predictions[0].argmax()])
+
+"""
+Awesome - looks like the model correctly classified the image.
+"""
+
+"""
+# Train a Classifier from Scratch
+
+![](https://storage.googleapis.com/keras-nlp/getting_started_guide/prof_keras_advanced.png)
+
+Now that we've gotten our hands dirty with classification, let's take on one
+last task: training a classification model from scratch!
+A standard benchmark for image classification is the ImageNet dataset, however
+due to licensing constraints we will use the CalTech 101 image classification
+dataset in this tutorial.
+While we use the simpler CalTech 101 dataset in this guide, the same training
+template may be used on ImageNet to achieve state of the art scores.
```

Comment:
"state-of-the-art"

The rule is that if an adjective contains multiple words, we put hypens between them so it's clear they are all modifiers for "scores"